### PR TITLE
Collection search builder

### PR DESCRIPTION
**Pull request by atz**
Salvaging work from #968

### CODE REVIEW

**Comment by atz** [comment]
@projecthydra/sufia-code-reviewers Rebased and re-ordered and reworked these commits so I can at least get some portion of improvements merged while polishing off the remainder.

**Comment by mjgiarlo** [comment]
:+1: 

Any thoughts on this, @tpendragon @escowles @jcoyne ?

File: app/search_builders/curation_concerns/collection_search_builder.rb [hunk 1]
```diff
@@ -1,35 +1,61 @@
...
+  #
   class CollectionSearchBuilder < ::SearchBuilder
     include FilterByType
     # Defines which search_params_logic should be used when searching for Collections
-    self.default_processor_chain = [:default_solr_parameters, :add_query_to_solr,
-                                    :add_access_controls_to_solr_params, :filter_models,
-                                    :some_rows, :sort_by_title]
+    # def initialize(context, access)
+    def initialize(context)
+      # @access = access_levels[access]
```
Comment (jcoyne) [inline app/search_builders/curation_concerns/collection_search_builder.rb]: Please remove this.

File: app/search_builders/curation_concerns/collection_search_builder.rb [hunk 2]
```diff
@@ -1,35 +1,61 @@
...
+  # @see https://github.com/projecthydra/curation_concerns/blob/master/app/search_builders/curation_concerns/README.md SearchBuilders README
+  # @note the default_processor_chain defined by Blacklight::Solr::SearchBuilderBehavior provides many possible points of override
+  #
   class CollectionSearchBuilder < ::SearchBuilder
     include FilterByType
     # Defines which search_params_logic should be used when searching for Collections
-    self.default_processor_chain = [:default_solr_parameters, :add_query_to_solr,
-                                    :add_access_controls_to_solr_params, :filter_models,
-                                    :some_rows, :sort_by_title]
+    # def initialize(context, access)
```
Comment (jcoyne) [inline app/search_builders/curation_concerns/collection_search_builder.rb]: Please remove this.

File: app/controllers/concerns/curation_concerns/collections_controller_behavior.rb [hunk 3]
```diff
@@ -13,6 +13,9 @@ module CollectionsControllerBehavior
 
       # This is needed as of BL 3.7
       copy_blacklight_config_from(::CatalogController)
+      configure_blacklight do |config|
+        config.search_builder_class = CurationConcerns::CollectionSearchBuilder
```
Comment (jcoyne) [inline app/controllers/concerns/curation_concerns/collections_controller_behavior.rb]: Is this change required? Why?

Comment (atz) [inline app/controllers/concerns/curation_concerns/collections_controller_behavior.rb]: If you want to include "Curations Concerns Collections Controller Behavior":
1. it should include a default `search_builder_class`
2. this is the only logical default value

Comment (jcoyne) [inline app/controllers/concerns/curation_concerns/collections_controller_behavior.rb]: But it never references that variable, so setting it is just code that isn't needed.

Comment (atz) [inline app/controllers/concerns/curation_concerns/collections_controller_behavior.rb]: I'm not sure what you mean.  `selects_collections.rb` already does the same thing.  `file_sets_controller_behavior.rb` and `curation_concern_controller.rb` both have methods for `search_builder_class`.  You're saying none of these do anything?

Comment (atz) [inline app/controllers/concerns/curation_concerns/collections_controller_behavior.rb]: @jcoyne: see here https://github.com/projecthydra/curation_concerns/blob/master/app/controllers/concerns/curation_concerns/selects_collections.rb#L7-L9

I'm happy enough to delete it, but it seems like it was required in that other case at least.  Should it be deleted in both?

**Comment by jcoyne** [comment]
@atz that module you were looking at was deprecated and unused.

**Comment by mjgiarlo** [comment]
@atz re: 
>  Still would appreciate comment about other `search_builder_class` assignment(s).
Why'd you dismiss @jcoyne's review if you want more review?

**Comment by jcoyne** [comment]
This PR also introduced a bunch of deprecation warnings here: https://github.com/projecthydra/curation_concerns/commit/3cb2320d6ec5693d04539e311eec3a514958479b

File: app/search_builders/curation_concerns/collection_search_builder.rb [hunk 4]
```diff
@@ -1,35 +1,59 @@
...
+  # @see https://github.com/projecthydra/curation_concerns/blob/master/app/search_builders/curation_concerns/README.md SearchBuilders README
+  # @note the default_processor_chain defined by Blacklight::Solr::SearchBuilderBehavior provides many possible points of override
+  #
   class CollectionSearchBuilder < ::SearchBuilder
     include FilterByType
     # Defines which search_params_logic should be used when searching for Collections
-    self.default_processor_chain = [:default_solr_parameters, :add_query_to_solr,
-                                    :add_access_controls_to_solr_params, :filter_models,
-                                    :some_rows, :sort_by_title]
+    def initialize(context)
```
Comment (jcoyne) [inline app/search_builders/curation_concerns/collection_search_builder.rb]: This changes the signature from 1-2 arguments to a single argument.  This is going to break sufia.
